contracts/FlowIDTableStaking.cdc

```diff
@@ -857,6 +857,70 @@ pub contract FlowIDTableStaking {
             }
         }
 
+        /// Enforces slot (count) limits for each node type (role). If more nodes
+        /// of a given type are registered than the limit, they are removed
+        /// randomly until the limits are met. When a node is removed, it's
+        /// committed tokens are transfered to it's unstaked bucket.
+        pub fun slotSelect() {
+            let allNodeIDs = FlowIDTableStaking.getNodeIDs()
+
+            let limits = [2, 2, 2, 2, 2]
+            var counts = [0, 0, 0, 0, 0]
+            for nodeID in allNodeIDs {
+                let nodeRecord = FlowIDTableStaking.borrowNodeRecord(nodeID)
+                if nodeRecord.initialWeight == 100 {
+                    counts[nodeRecord.role - 1] = counts[nodeRecord.role - 1] + 1
+                }
+            }
+
+            for index, element in counts {
+                while counts[index] > limits[index] {
```

Comment:
You get the element at the index in each iteration of the loop, so there is no need to access the array again. you can just do `while element > limits[index]`

Reply:
As mentioned above, the "element" parameter is going away in the change from arrays to dictionaries. But I have a question. As I understand it, "element" would have been a constant within the loop body which means I can't use it to update the values in "counts". Obviously I need to be updating the counts as we're removing nodes. It's not a critical question anymore, just confirming my understanding.

Reply:
Oh and as I'm updating counts I also need to be consuming the latest updated values. If "element" is constant, it wouldn't work for that either.